spark3/ethereum/condition.py

```diff
@@ -0,0 +1,110 @@
+                col_name == self.key_alias and col_type == self.key_type]) != 1:
+            raise ColumnNotFoundInDataFrame(f'{self.key_alias}({self.key_type})', df)
+
+        return df.filter(col(self.key_alias) == self.transform(raw))
+
+
+def return_self_condition(alias: str, _type: str) -> Condition[str]:
+    return Condition(key_alias=alias, key_type=_type, transform=lambda x: x)
+
+
+def return_hash_condition(alias: str, _type: str) -> Condition[str]:
+    return Condition(key_alias=alias, key_type=_type, transform=lambda x: abs(hash_unsafe_bytes(x)) % 10)
+
+
+class Conditions:
+    def __init__(self,
+                 ctype: str,
+                 address_condition: Condition[str],
+                 address_hash_condition: Optional[Condition[str]] = None,
+                 selector_condition: Optional[Condition[Dict[str, any]]] = None,
+                 selector_hash_condition: Optional[Condition[Dict[str, any]]] = None):
+        if ctype != 'function' and ctype != 'event':
+            raise TypeError('The column type must be "function" or "event"')
+
+        self.ctype = ctype
+        self.address_condition = address_condition
+        self.address_hash_condition = address_hash_condition
+        self.selector_condition = selector_condition
+        self.selector_hash_condition = selector_hash_condition
+
+    def act(self, df: DataFrame, address: str, single_abi: Dict[str, any]) -> DataFrame:
+        result_df = self.address_condition.acts(df, address)
+
+        if self.address_hash_condition is not None:
+            result_df = self.address_hash_condition.acts(result_df, address)
+        if self.selector_condition is not None:
+            result_df = self.selector_condition.acts(result_df, single_abi)
+        if self.selector_hash_condition is not None:
+            result_df = self.selector_hash_condition.acts(result_df, single_abi)
+
+        return result_df
+
+    @staticmethod
+    def new_datawave_trace_condition():
+        default_trace_function_selector_condition = Condition(
+            key_alias='selector',
+            key_type='string',
+            # :paras x: function abi dict
+            transform=lambda x: encode_hex(function_abi_to_4byte_selector(x))
+        )
+
+        default_trace_function_hash_condition = Condition(
+            key_alias='selector_hash',
+            key_type='int',
+            # :paras x: function abi dict
+            transform=lambda x: abs(hash_unsafe_bytes(encode_hex(function_abi_to_4byte_selector(x)))) % 10
+        )
+
+        return Conditions(
+            ctype='function',
+            address_condition=return_self_condition('to_address', 'string'),
+            address_hash_condition=return_hash_condition('address_hash', 'int'),
+            selector_condition=default_trace_function_selector_condition,
+            selector_hash_condition=default_trace_function_hash_condition
+        )
+
+    @staticmethod
+    def new_datawave_log_condition():
```

Comment:
I guess those Conditions better be moved outside this client library.

And let the library users (like datawaves & @raycursive ) implement them : )